pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -361,7 +431,36 @@ func extractLibrariesFromAnnotations(
 	return libList
 }
 
-func injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo, autoDetected bool, injectionType string) error {
+// isEnabled indicates if Single Step Instrumentation is enabled for the namespace in the cluster
+func (apmInstrumentation *APMInstrumentation) isEnabled(namespace string) bool {
+	apmInstrumentationEnabled := config.Datadog.GetBool("apm_config.instrumentation.enabled")
+
+	if !apmInstrumentationEnabled {
+		log.Debugf("APM Instrumentation is disabled")
+		return false
+	}
+
+	return !apmInstrumentation.filter.IsExcluded(nil, "", "", namespace)
+}
+
+// shouldInject returns true if Admission Controller should inject standard tags, APM configs and APM libraries
+func (apmInstrumentation *APMInstrumentation) shouldInject(pod *corev1.Pod) bool {
```

Comment:
Note for reviewers: I believe this might be redundant if the label selectors are properly set.
For now I've just moved this function from `pkg/clusteragent/admission/mutate/common.go` and adapted it to take into account if apm instrumentation is enabled.
In a future PR we can evaluate if these functions are really needed.

Reply:
Label selector takes into account CA configuration only, not labels/annotations set on the pod.
shouldInject(...) from common.go is also used to make a decision to injects tags and configs and is independent from auto instrumentation (https://github.com/DataDog/datadog-agent/blob/main/pkg/clusteragent/admission/mutate/tags.go#L78-L81). With that, I'm not sure it makes sense to move this function from https://github.com/DataDog/datadog-agent/blob/main/pkg/clusteragent/admission/mutate/common.go#L212-L226.

Reply:
I didn't move the function, I copied it and adapted it. Sorry, I was not precise in my first message.

There are 2 `shouldInject` functions now: one in `auto_instrumentation.go` and another in `common.go`. The only difference is that the one in `auto_instrumentation.go` checks `isApmInstrumentationEnabled` and the other one doesn't, because as you mentioned the other use cases (inject tags and inject configs) are independent from apm instrumentation.

As you mentioned in another comment, I agree that this can be improved. I prefer to leave it like this in this PR because I just wanted to focus on the filtering part, but I'm going to address this kind of duplications in another PR that I'm preparing.

Reply:
I think I might have misunderstood something.

In the code in the `main` branch if then pod is not labelled and "mutate_unlabelled" is set to false, we still inject tags and configs if apm instrumentation is enabled (because `shouldInject` returns true). Is that correct?

I thought that APM was independent from the other two. Does APM need the injected tags and configs to work?

Reply:
Summarizing what we have discussed during the sync: SSI requires configs and tags together with tracing libraries injection. Right now, even if webhooks for configs and tags are disabled, apm_instrumentation webhook will inject config and tags env variables.

Reply:
I've added a new commit that fixes this.
The behavior should now be the same as it was before this PR.